src/rtc/h5_prep.py

```diff
@@ -371,11 +557,52 @@ def get_metadata_dict(product_id: str,
             ['config_files', cfg_in.run_config_path,
              'List of input config files used'],
         'RTC/metadata/processingInformation/inputs/demSource':
-            ['dem_source', dem_description, 'DEM source description']
+            ['dem_source', dem_description, 'DEM source Description'],
     }
+
+    # Add reference to the thermal noise correction algorithm when the correction is applied
+    if cfg_in.groups.processing.apply_thermal_noise_correction:  # 3.3
+        metadata_dict['RTC/metadata/processingInformation/noiseRemovalAlgorithmReference'] =\
+        ['noise_removal_algorithm_reference',
+         ('https://sentinels.copernicus.eu/documents/247904/2142675/'
+          'Thermal-Denoising-of-Products-Generated-by-Sentinel-1-IPF.pdf/'
+          '11d3bd86-5d6a-4e07-b8bb-912c1093bf91?t=1511973926000'),
+          'A reference to the noise removal algorithm applied']
+
+    # Add RTC algorithm reference depending on the algorithm applied
+    if cfg_in.groups.processing.apply_rtc: # 3.4
+        if cfg_in.groups.processing.rtc.algorithm_type == 'area_projection':
+            url_rtc_algorithm_document = 'https://ieeexplore.ieee.org/document/9695438'
+        elif cfg_in.groups.processing.rtc.algorithm_type == 'bilinear_distribution':
+            url_rtc_algorithm_document = 'https://ieeexplore.ieee.org/document/5752845'
+        else:
+            raise NotImplementedError
+        metadata_dict['RTC/metadata/processingInformation/rtcAlgorithmReference'] =\
+                ['rtc_algorithm_reference',
+                 url_rtc_algorithm_document,
+                'A reference to the RTC algorithm applied']
```

Comment:
Same idea here. Add something like "RTC not applied" if `cfg_in.groups.processing.apply_rtc` is `False`